source/byte_buf.c

```diff
@@ -41,9 +40,9 @@ int aws_byte_buf_init(struct aws_byte_buf *buf, struct aws_allocator *allocator,
 }
 
 int aws_byte_buf_init_copy(struct aws_byte_buf *dest, struct aws_allocator *allocator, const struct aws_byte_buf *src) {
-    if (!allocator || !dest || !aws_byte_buf_is_valid(src)) {
-        return aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
-    }
+    AWS_ERROR_PRECONDITION(allocator);
+    AWS_ERROR_PRECONDITION(dest);
```

Comment:
the IS_VALID below can probably stay, but these guys probably should assert too.

Reply:
Done